First solution to Christian's project

juliaquach02 opened the pull request:
Hi all, 

this is my first solution to Christian's project and I am trying to open a pull request.

I think [this](https://github.com/Dr-Eberle-Zentrum/Advanced-data-processing-with-R/blob/julia_tries_christians_project/Projects/crudi3/juliaquach02.md) is the link to .md file.

@crudi3  Are you satisfied with this solution?

Best
Julia

martin-raden commented:
Hi Julia,

na das sieht doch schon sehr gut aus! Und auch super dokumentiert. Genau dazu sind die markdown notebooks da... 👍

Die inhaltlichen Kommentare auch zu den Plotlayouts überlasse ich mal Christian. Ich hab mal etwas genauer auf den Quellcode geschaut.

Hier ein paar Punkte die mir aufgefallen sind.

- [ ] anstatt die Spalten explizit zu listen (für das `select` würde ich das Entwickeln einer entsprechenden regex vorschlagen, das alle spaltennamen, die du haben willst, beschreibt. die regex kannst du dann im `select()` mit [`matches()`](https://tidyselect.r-lib.org/reference/starts_with.html) direkt verwenden
- [ ] damit dein spaltenrename via `colnames()` dann noch geht, musst du ggf. die `ingle` spalte via `relocate()` nach vorn holen.. 
- [ ] deine `result` Berechnung lässt sich auf 3 Zeilen eindampfen, wenn du "im tidyverse Stil" rangehst. Im moment ist deine `df` Tabelle im wide table format und damit nicht "tidy", weil in jeder Zeile mehrere trust level Zahlen vorliegen. was du brauchst, um mit tidyverse darauf effizient rumzurödeln ist eine Tabelle, die 3 spalten hat: Inglehart-Index, Institution, Trust-Level. Die kannst du ganz einfach erzeugen, indem du die wide table tabelle via [`pivot_longer()`](https://r4ds.had.co.nz/tidy-data.html#pivoting) ins long-table Format überführst... anschliessend kannst du mit [grouping und `summarize()`](https://r4ds.had.co.nz/transform.html#grouped-summaries-with-summarise) direkt die Mittelwerte pro Institution bestimmen. sprich `df %>% privot_longer(..) %>% group_by(..) %>% summarize(...) -> result`  ! versuchs mal! 😀
- [ ] gleiches hilft dir nämlich dann auch bei deinem letzten coding: das grouping kann auch auf zwei spalten basieren, sprich Inglehart-Index UND Institution.. damit bekommst du dann direkt die Mittelwerte pro Kombination raus, etc...
- [ ] bin insgesamt allerdings am überlegen, ob der letzte Graph wirklich hilfreich ist, die 4 Gruppen zu vergleichen, oder ob 4-bars-per-institution besser wären, um die trends zu vergleichen. kannst du ja ggf. mal versuchen?!
- [ ] `means <- means[, c(5, 1:4)]` ist auch ein Fall für `relocate()` (wenn man will)
- [ ] `groups[[i]][, 1:ncol( groups[[i]] ) ]` ist das nicht doppelmoppel? sprich das zweite subsetting tut nichts?!
- [ ] genauso hier `colnames( df )[1:length(df)]` ?!

Alles in allem: wilde Sache die da gebaut hast! 😜👍

Denke man könnte einige Stellen kompaktieren, indem man mehr Richtung und mit tidyverse denkt/arbeitet. Sprich gerade grouping und co sind extrem hilfreich für derartige Sachen und machen viele for-loops und zwischendatenstrukturen unnötig. Allerdings muss man dazu immer die daten erstmal "tidy" machen und dann seine "Programmierdenke" etwas umstellen. Aber lohnt sich, aus meiner Sicht, total!

Hoffe das hilft.
Liebe Grüsse,
Martin

crudi3 commented:
Hey @juliaquach02 , 
sorry für die späte Antwort... Die Lösung sieht super aus, hab es mir tatsächlich genauso vorgestellt!

Zwei Dinge sind mir dann aber noch aufgefallen. Ich hätte es noch cool gefunden, wenn du die größte Gruppendifferenz (Sieht so aus als wäre das das Gesundheitswesen) und die kleinste (homogenste) Gruppendifferenz markiert hättest. Sprich dort wo die post-/materialistische Einstellung den größten und den geringsten Unterschied macht. Das zweite ist reine Kontinuitätssache, nämlich die Beschriftung "Mean Trust Level" sollte natürlich, wenn der Rest des Diagramms auf deutsch ist, dann auch auf deutsch sein ;) Aber das hier ist schon eher nörgeln auf hohem Niveau, also ich sehe das Projekt zu 100% erfüllt!

Inhaltlich bin ich tatsächlich recht überrascht, dass die Gruppen alle recht ähnlich sind. Ich hätte da mit etwas größeren Unterschieden gerechnet. Wenn man sich da in die Literatur zu Post-/Materialismus reinfuchst, wird immer wieder beschrieben, dass Postmaterialisten höhere zwischenmenschliche und institutionelle Vertrauenslevel vorweisen und dass die Wertekomplexe der postmaterialistisch angelegten Personen für demokratische Institutionen und die allgemeine demokratische Kultur extrem wichtig seien. Daher bin ich etwas verwundert über die doch recht homogenen Vertrauenslevel der einzelnen Gruppen. Bis auf das Gesundheitswesen (dem Materialisten sogar eher vertrauen als Postmaterialisten!) sind die Gruppen tatsächlich sehr gleich. Hätte ich so nicht gedacht!

An der Stelle nochmal danke für die doch sehr schnelle und gute Bearbeitung des Projekts und nochmal Entschuldigung für die späte Rückmeldung!
Viele Grüße,
Christian

juliaquach02 commented:
Hi there,

you can find an updated .md-file [here]( ).

@martin-raden thank your for your valuable tips! 
The functions pivot_longer() and grouping were indeed very useful.  For some steps in the code, I could still only think of a for-loop but overall the code became much shorter. 

@crudi3 Thank you for your kind words! That was nice to hear. Following, I will address your two comments:

> Ich hätte es noch cool gefunden, wenn du die größte Gruppendifferenz (Sieht so aus als wäre das das Gesundheitswesen) und die kleinste (homogenste) Gruppendifferenz markiert hättest. Sprich dort wo die post-/materialistische Einstellung den größten und den geringsten Unterschied macht.

Sure, for the minimum and maximum difference between the groups, I added text annotations to the plot. That was the clearest label that I could think of but I am open to suggestions.

> Das zweite ist reine Kontinuitätssache, nämlich die Beschriftung "Mean Trust Level" sollte natürlich, wenn der Rest des Diagramms auf deutsch ist, dann auch auf deutsch sein ;) 

Ah yes, that is true. I changed the y-axis lable to "Vertrauenslevel". I am not sure whether this is the right word but I can quickly change it if you have a better translation.

Also thank you for the background information. That was interesting to me. :)

Best
Julia
